spec/syntax.md

```diff
@@ -705,6 +705,36 @@ on the pairing, ordering, or contents of _markup_ during _formatting_.
 Examples in this section are meant to be illustrative and
 might not match future requirements or usage.
 
+> [!NOTE]
+> The Tech Preview does not provide a built-in mechanism for overriding
+> values in the _formatting context_ (most notably the locale)
+> Nor does it provide a mechanism for identifying specific expressions
+> such as by assigning a name or id.
+> The utility of these types of mechanisms has been debated.
+> There are at least two proposed mechanisms for implementing support for
+> these. 
+> Specifically, one mechanism would be to reserve specifically-named options, 
+> possibly using a Unicode namespace (i.e. `locale=xxx` or `u:locale=xxx`).
+> Such options would be reserved for use in any and all functions or markup.
+> The other mechanism would be to use the reserved "expression attribute" syntax
+> for this purpose (i.e. `@locale=xxx` or `@id=foo`)
+> Neither mechanism was included in this Tech Preview.
+> Feedback on the preferred mechanism for managing these features
+> is strongly desired.
+> 
+> In the meantime, function authors and other implementers are cautioned to avoid creating 
+> function-specific or implementation-specific option values for this purpose.
+> One workaround would be to use the implementation's namespace for these 
+> features to insure later interoperability when such a mechanism is finalized 
+> during the Tech Preview period.
+> Specifically:
+> - Avoid specifying an option for setting the locale of an expression as different from
+>   that of the overall _message_ locale, or use a namespace that later maps to the final
+>   mechanism.
+> - Avoid specifying an option to link opening and closing markup expressions
+>   such as a `startRef` on closing markup or matching `id` or `ref` values in paired
+>   markup.
```

Comment:
The intent with the current language is to direct away from overall behaviour, than any specific implementation for such behaviour. Does that not also cover your concern?

Reply:
But using function options for this will always be possible anyways. I think the note should only say: "we're still thinking about this; if you need it now and options are sufficient, please use a namespace; also, give us feedback". It should not say: "avoid doing it".

Reply:
@stasm We agreed to this language yesterday, but I can see what you're getting at. However, I need concrete proposals...

\`\`\`suggestion
(goes here 😸 )
\`\`\`

Reply:
Sorry for not being more specific in my earlier comment. How about the following, for consistency with the previous bullet:

```suggestion
> - Avoid specifying an option to link opening and closing markup expressions
>   such as a `startRef` on closing markup or matching `id` or `ref` values in paired
>   markup, or use a namespace that later maps to the final mechanism.
```

Reply:
I'm finding it hard to understand what the proposed addition means. Would this namespace be on the markup names or the options? So is this about forbidding `foo:startRef` and `foo:id` in addition to the un-namespaced options, or something else?

Reply:
It’s about recommending namespaced markup options. This is perhaps my not speaking English natively, but what I want to say is “avoid specifying… or use namespaces if you need it”. What I don’t want to say is: “avoid specifying… and avoid using…”

Reply:
How about:

```suggestion
> - Avoid specifying options for the purpose of linking expressions 
>   (such as to pair opening markup to closing markup) 
>   or require that such options use an implementation-specific namespace
>   while cautioning users that such option might not be replaced or unavailable later.
```

Reply:
Yes, thank you!